nats-ballerina/tests/tls_connection_tests.bal

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+@test:Config {
+    groups: ["nats-basic"]
+}
+public isolated function testTlsConnection1() {
+    SecureSocket secured = {
+        cert: {
+            path: "tests/certs/truststore.p12",
+            password: "password"
+        },
+        key: {
+            path: "tests/certs/keystore.p12",
+            password: "password"
+        },
+        protocol: {
+            name: TLS
+        }
+    };
+    Client|Error newClient = new("nats://localhost:4225", secureSocket = secured);
+    if (newClient is error) {
+        test:assertFail("NATS Connection initialization with TLS failed.");
+    }
+}
+
+@test:Config {
+    groups: ["nats-basic"]
+}
+public isolated function testTlsConnection2() {
```

Comment:
Isn't this test and testTlsConnection4 test tests the same?

Reply:
This one has the keystore config as well,`testTlsConnection4` only has trustore config

Reply:
Keystore is needed when we want to validate the client certificate from the server-side.
As per the given configurations here for the NATS server(https://github.com/ballerina-platform/module-ballerinax-nats/pull/163/files#diff-c5b26782d4f7a55482cfef544ef4d35aa7c2c1880813fc941089688f29e06d20R1) this will not do that client validation. right?
As I saw in this documentation, https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro/tls_mutual_auth#validating-a-client-certificate I think it will be applicable only if we say `verify:    true`.
```
tls {
  cert_file: "server-cert.pem"
  key_file:  "server-key.pem"
  ca_file:   "rootCA.pem"
  verify:    true
}
```

There is no harm in configuring the Keystore. It'll get ignored. So basically this will do the same thing as in `testTlsConnection4`.

Reply:
Added the keystore so that keystore part of the java code will be covered. Since that is the effort 😅

Reply:
Fixed in 3bce0d3b0839d40529af79728720336c3c77ebb3

Reply:
@Bhashinee @aashikam I guess then we need a test for mTLS ?

Reply:
Added in 3ffca1c8a0b659cbe7f78ba4b87037cb269f5490